### PR TITLE
Feature/tg 125/unify code backend

### PR DESCRIPTION
**Pull request by CristianBucci**
- Unify endpoints response and message
- Change endpoints to get by params
- Change joi validation regex in some properties to accept spaces between words
- Add ADMIN role in getAllAdmins route

QA: 
Test in Postman each entity endpoint:
- getAll
- getById
- getBy'Params'
- create
- edit
- delete

Note: 'Params' could be by: name, lastName, email, phone, etc...

### CODE REVIEW

File: src/controllers/employee.js [hunk 1]
```diff
@@ -1,58 +1,44 @@
...
+    const { id } = req.params;
+    const employee = await Employee.findById(id);
+    if (!employee) {
       // eslint-disable-next-line no-throw-literal
       throw {
         message: 'Employee not found.', status: 404,
       };
+    } else {
+      return res.status(200).json({
+        message: 'Employees found.',
```
Comment (Ansitox) [inline src/controllers/employee.js:36]: Employee* found

File: src/controllers/projects.js [hunk 2]
```diff
@@ -1,35 +1,44 @@
...
       populate: {
         path: 'employeeId',
       },
     });
-    if (!projects) {
+
+    if (!projects.length) {
       // eslint-disable-next-line no-throw-literal
       throw {
         message: 'Project not found.', status: 404,
```
Comment (Ansitox) [inline src/controllers/projects.js:15]: No projects found*

File: src/controllers/projects.js [hunk 3]
```diff
@@ -1,35 +1,44 @@
...
-      return res.status(200).json({
-        message: 'Projects found.',
-        data: projects,
-      });
-    }
-    let filterByParams;
-    if (req.query.name) {
-      filterByParams = projects.filter((project) => project.name === req.query.name);
+    return res.status(200).json({
+      message: 'Project found',
```
Comment (Ansitox) [inline src/controllers/projects.js:19]: Projects* found

File: src/controllers/superAdmin.js [hunk 4]
```diff
@@ -1,36 +1,19 @@
...
   try {
-    const superAdmins = await SuperAdmin.find();
-    if (!superAdmins) {
+    const superAdmin = await SuperAdmin.find(req.query);
+
+    if (!superAdmin.length) {
       // eslint-disable-next-line no-throw-literal
       throw {
-        message: 'Super admin not found.', status: 404,
+        message: 'Super Admin not found.', status: 404,
```
Comment (Ansitox) [inline src/controllers/superAdmin.js:11]: No Super Admins found*

File: src/controllers/admins.js [hunk 5]
```diff
@@ -2,34 +2,18 @@ import firebaseApp from '../helpers/firebase/index';
...
 const getAllAdmins = async (req, res) => {
-  const queriesArray = Object.keys(req.query);
   try {
     const admins = await Admins.find(req.query);
-    if (!admins) {
+
+    if (!admins.length) {
       // eslint-disable-next-line no-throw-literal
       throw {
         message: 'Admin not found.', status: 404,
```
Comment (Ansitox) [inline src/controllers/admins.js:11]: No Admins found

File: src/controllers/timeSheet.js [hunk 6]
```diff
@@ -1,129 +1,32 @@
...
       .populate('task')
       .populate('employee')
       .populate('project');
-    if (!timesheetFound || !timesheetFound.length) {
+    if (!timesheets.length) {
       // eslint-disable-next-line no-throw-literal
       throw {
-        message: 'Timesheet not found.',
-        status: 404,
+        message: 'Timesheet not found.', status: 404,
```
Comment (Ansitox) [inline src/controllers/timeSheet.js:12]: No Timesheets* found

**Review by Ansitox — COMMENTED** [review]
I think we should change the update and delete messages that contain IDs to not showing them.

**Review by Ansitox — APPROVED** [review]
Looks good to me

**Review by julianpiaggio — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK

We wanted to tell you that we consider that your pr complies
the requirements to be approved, and that it can be merged into the master branch 👍

**Pull request merged**